src/subpages/list.tsx

```diff
@@ -1,263 +1,138 @@
-    } else {
-      setSearchParams({ ...searchParams, offset });
-    }
-  };
-  const olderHandler = () => {
-    setOffset(+offset + limit);
-  };
-  const newerHandler = () => {
-    setOffset(+offset - limit);
-  };
+const ListPage: React.FC<ListPageProps> = ({ location }) => {
+  const [offset, olderHandler, newerHandler] = useOffset(location);
   const [excludeEmptyTxs, setExcludeEmptyTxs] = useState(false);
   return (
-    <div>
+    <main>
       <Checkbox
         label="Include blocks having any tx"
         checked={excludeEmptyTxs}
-        onChange={(_, checked) => {
-          setExcludeEmptyTxs(!!checked);
-        }}
+        onChange={() => setExcludeEmptyTxs(!excludeEmptyTxs)}
       />
-      <BlockListComponent
-        variables={{ offset, limit, excludeEmptyTxs }}
-        pollInterval={POLL_INTERVAL}>
+      <BlockListComponent variables={{ offset, limit, excludeEmptyTxs }}>
         {({ data, loading, error }) => {
-          if (error) return <p>error!</p>;
-
-          const timestamps: Date[] | null =
-            data && data.blockQuery && data.blockQuery.blocks
-              ? data.blockQuery.blocks.map(block => new Date(block!.timestamp))
-              : null;
-
-          let interval: number | null = timestamps ? 0 : null;
-          if (interval != null && timestamps) {
-            for (let i = 0; i < timestamps.length - 1; i++) {
-              interval += +timestamps[i] - +timestamps[i + 1];
-            }
-            interval /= (timestamps.length - 1) * 1000;
+          if (error) {
+            console.error(error);
+            return <p>{error.message}</p>;
           }
-
-          const difficulties: number[] | null =
+          const blocks =
             data && data.blockQuery && data.blockQuery.blocks
-              ? data.blockQuery.blocks.map(block => block!.difficulty)
+              ? (data.blockQuery.blocks as Block[])
               : null;
-          let difficulty = 0;
-          if (difficulty != null && difficulties) {
-            difficulty =
-              difficulties.reduce((d, sum) => d + sum, 0) / difficulties.length;
-          }
 
-          const txNumbers: number[] | null =
-            data && data.blockQuery && data.blockQuery.blocks
-              ? data.blockQuery.blocks.map(block => block!.transactions.length)
-              : null;
-          let totalTxNumber = 0;
-          if (difficulty != null && difficulties) {
-            totalTxNumber = txNumbers.reduce((a, b) => a + b, 0);
-          }
+          const timeTaken: IColumn = {
+            key: 'columnTimeTaken',
+            name: 'Time Taken',
+            minWidth: 50,
+            maxWidth: 200,
+            ...commonProps,
+            isSortedDescending: true,
+            data: 'string',
+            isPadded: true,
+            onRender: (block: Block, index) => {
+              if (blocks === null || index === undefined)
+                throw Error('blocks is null');
```

Comment:
I wonder the case when `index === undefined` became true.

Reply:
It just type guard. index can be undefined.

<img width="693" alt="Screen Shot 2020-08-12 at 4 50 44 PM" src="https://user-images.githubusercontent.com/5278201/89989480-f83d8d80-dcbb-11ea-9dff-60847203477f.png">

Reply:
Then the error message seems not to suit to this case because it will be always `'blocks is null'` even if index is `undefined`.